vng_api_common/middleware.py

```diff
@@ -96,6 +101,111 @@ def has_scopes(self, scopes: Union[Scope, None]) -> bool:
         return scopes.is_contained_in(scopes_provided)
 
 
+class JWTAuth:
+    def __init__(self, encoded: str = None):
+        self.encoded = encoded
+
+    @property
+    def applicaties(self) -> Union[list, None]:
+        if self.client_id is None:
+            return []
+
+        applicaties = self._get_auth()
+
+        if not applicaties:
+            auth_data = self._request_auth()
+            applicaties = self._save_auth(auth_data)
+
+        return applicaties
+
+    def _request_auth(self) -> list:
+        client = AuthorizationsConfig.get_client()
+
+        response = client.list(
+            'applicatie',
+            query_params={'client_ids': self.client_id}
+        )
+        return underscoreize(response['results'])
+
+    def _get_auth(self):
+        return Applicatie.objects.filter(client_ids__contains=[self.client_id])
+
+    def _save_auth(self, auth_data):
```

Comment:
I definitely want this in a database transaction

Reply:
`@transaction.atomic` is added